phoenix-core-client/src/main/java/org/apache/phoenix/jdbc/HighAvailabilityPolicy.java

```diff
@@ -66,14 +67,19 @@ private void transitStandby(HighAvailabilityGroup haGroup, String zkUrl)
                     zkUrl, haGroup.getGroupInfo());
             ConnectionQueryServices cqs = null;
             try {
-                cqs = PhoenixDriver.INSTANCE.getConnectionQueryServices(
-                        haGroup.getGroupInfo().getJDBCUrl(zkUrl), haGroup.getProperties());
-                cqs.closeAllConnections(new SQLExceptionInfo
-                        .Builder(SQLExceptionCode.HA_CLOSED_AFTER_FAILOVER)
-                        .setMessage("Phoenix connection got closed due to failover")
-                        .setHaGroupInfo(haGroup.getGroupInfo().toString()));
-                LOG.info("Closed all connections to cluster {} for HA group {}", zkUrl,
-                        haGroup.getGroupInfo());
+                //Close connections for every HAURLInfo's (different principal) connections for a give HAGroup
+                for (HAURLInfo haurlInfo : HighAvailabilityGroup.URLS.get(haGroup.getGroupInfo())) {
+                    String jdbcZKUrl = haGroup.getGroupInfo().getJDBCUrl(zkUrl, haurlInfo);
+                    cqs = PhoenixDriver.INSTANCE.getConnectionQueryServices(
+                            jdbcZKUrl, haGroup.getProperties());
+                    cqs.closeAllConnections(new SQLExceptionInfo
+                            .Builder(SQLExceptionCode.HA_CLOSED_AFTER_FAILOVER)
+                            .setMessage("Phoenix connection got closed due to failover")
+                            .setHaGroupInfo(haGroup.getGroupInfo().toString()));
+                    LOG.info("Closed all connections to cluster {} for HA group {}",
+                            jdbcZKUrl, haGroup.getGroupInfo());
+                }
+
```

Comment:
Shouldn't the finally be inside the loop?

Reply:
Yes Thanks for catching that